src/fairness/weighted_tree/test/load_accounting_db.cpp

```diff
@@ -0,0 +1,291 @@
+                                        "WHERE parent_bank=?";
+    rc = sqlite3_prepare_v2 (DB,
+                             select_sub_banks_stmt.c_str (),
+                             -1,
+                             &b_select_sub_banks_stmt,
+                             0);
+
+    // SELECT statement to get all associations from a bank
+    std::string select_associations_stmt = "SELECT association_table.username, "
+                                           "association_table.shares, "
+                                           "association_table.bank, "
+                                           "association_table.job_usage "
+                                           " FROM association_table "
+                                           " WHERE association_table.bank=?";
+    rc = sqlite3_prepare_v2 (DB,
+                             select_associations_stmt.c_str (),
+                             -1,
+                             &b_select_associations_stmt,
+                             0);
+
+    // SELECT statement to get the root bank from the bank table
+    std::string select_root_bank_stmt = "SELECT bank_table.bank "
+                                        "FROM bank_table "
+                                        "WHERE parent_bank=''";
+    // compile SQL statement into byte code
+    rc = sqlite3_prepare_v2 (DB,
+                             select_root_bank_stmt.c_str(),
+                             -1,
+                             &b_select_root_bank_stmt,
+                             0);
+    if (rc != SQLITE_OK) {
+        fprintf (stderr, "Failed to fetch data: %s\n", sqlite3_errmsg (DB));
+        sqlite3_close (DB);
+
+        exit (EXIT_FAILURE);
+    }
+
+    rc = sqlite3_step (b_select_root_bank_stmt);
+    // store root bank name
+    std::string root_bank;
+    if (rc == SQLITE_ROW) {
+        root_bank = reinterpret_cast<char const *> (
+            sqlite3_column_text (b_select_root_bank_stmt, 0));
+    }
+    // otherwise, there is either no root bank or more than one
+    // root bank; the program should exit
+    else {
+        std::cerr << "root bank not found, exiting" << std::endl;
+        exit (EXIT_FAILURE);
+    }
+
+    // call recursive function
+    std::shared_ptr<weighted_tree_node_t> root = nullptr;
+    root = get_sub_banks (DB,
+                          root_bank,
+                          nullptr,
+                          b_select_shares_stmt,
+                          b_select_sub_banks_stmt,
+                          b_select_associations_stmt);
+
+    // destroy the prepared SQL statements
+    sqlite3_finalize (b_select_root_bank_stmt);
+    sqlite3_finalize (b_select_shares_stmt);
+    sqlite3_finalize (b_select_sub_banks_stmt);
+    sqlite3_finalize (b_select_associations_stmt);
+
+    weighted_walk_t walker (root);
+    walker.run ();
+
+    const auto &users = walker.get_users ();
+
+    // close DB connection
+    sqlite3_close (DB);
```

Comment:
Minor nit: I'd move this up a few lines to just below the sqlite3_finalize calls